### DISCUSSION TITLE
Any plans to introduce API similar to Akka Typed from scala?

### DISCUSSION THREAD

**Discussion by adnang** [akkadotnet/akka.net#6267] [General]
If not, why not? Would there be any limitations in .NET preventing it?

**Comment by coldwarrl** [comment]
To my knowledge, this is planned for version 2.0

**Reply by adnang** [reply]
I see, thanks! are there any preliminary docs/discussions about it?

**Reply by coldwarrl** [reply]
It was discussed in one of the community meetings, https://github.com/akkadotnet/akka.net/discussions/categories/community-standups, pretty long ago. I do not remember the concrete meeting instance, maybe @Aaronontheweb has more details...

**Reply by Aaronontheweb** [reply]
Yep, did that here: https://youtu.be/j2nNRO4pVMI?t=3289 - about two years ago.

Still want to do all of that, but at the moment we're still chugging away on Akka.NET v1.5 plus some other projects. Typed actors solve one of the most painful problems with Akka.NET which is cohesion - it's very much top of mind for us to address.

**Comment by Aaronontheweb** [comment]
Definitely - we discussed some hypotheticals around doing this a couple of years ago: https://youtu.be/j2nNRO4pVMI?t=3289
